### PR TITLE
Replace string version of UUID with 16 byte slice and remove the old,

### PR DESCRIPTION
**Pull request by spencerkimball**
inefficient md5 computation, which was just stupid.

### CODE REVIEW

**Comment by petermattis** [comment]
Looks like http://code.google.com/p/go-uuid/uuid no longer exists. I'm not sure if it moved to github somewhere or not. We could always fork.

**Comment by petermattis** [comment]
LGTM modulo the go-uuid problem.

**Comment by spencerkimball** [comment]
Does this mean you get go get cockroach anymore?

**Comment by petermattis** [comment]
Yeah, `go get github.com/cockroachdb/cockroach` would probably not work right now.

File: proto/data.go [hunk 1]
```diff
@@ -443,9 +445,15 @@ func MakePriority(r *rand.Rand, userPriority int32) int32 {
 	return math.MaxInt32 - rand.Int31n(math.MaxInt32/userPriority)
 }
 
-// MD5Equal returns whether the specified md5.Size byte arrays are equal.
-func MD5Equal(a, b [md5.Size]byte) bool {
-	for i := 0; i < md5.Size; i++ {
+// TxnIDEqual returns whether the transaction IDs are equal.
+func TxnIDEqual(a, b []byte) bool {
+	if a == nil && b == nil {
```
Comment (tschottdorf) [inline proto/data.go]: why not scratch everything here up to the for loop before and then at the end return `len(a) == len(b)` instead?

**Comment by tschottdorf** [comment]
LGTM

File: proto/data.go (hunk 1 above)
Comment (spencerkimball) [inline proto/data.go]: Good point. It's been fixed up in the PR which replaces UUID.

**Pull request merged**